python/ray/experimental/workflow/common.py

```diff
@@ -59,13 +72,31 @@ def execute(self) -> RRef:
         # proper context. To prevent it, we put it inside a tuple.
         step_inputs = (self._input_placeholder, workflow_outputs,
                        self._input_object_refs)
-        output = self._step_execution_function(self._step_id, step_inputs)
+        output = self._step_execution_function(self._step_id, step_inputs,
+                                               forward_output_to)
         if not isinstance(output, WorkflowOutputType):
             raise TypeError("Unexpected return type of the workflow.")
         self._output = output
         self._executed = True
         return output
 
+    def _visit_workflow_dag(self, visited_workflows: Set["Workflow"]):
```

Comment:
`_get_dag`

And don't pass the set from outside, instead, have a helper function for this.

Btw, are we actually getting the dag here? I feel it's like adding all nodes in the dag into some set.